plutus-core/testlib/PlutusCore/Generators/PIR/Common.hs

```diff
@@ -0,0 +1,74 @@
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE LambdaCase            #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE TypeApplications      #-}
+{-# LANGUAGE UndecidableInstances  #-}
+
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module PlutusCore.Generators.PIR.Common where
+
+import Data.Bifunctor
+import Data.Map.Strict (Map)
+import Data.Map.Strict qualified as Map
+import GHC.Stack
+import Test.QuickCheck.Modifiers (NonNegative (..))
+import Test.QuickCheck.Property
+import Text.Pretty
+import Text.PrettyBy
+import Text.PrettyBy.Internal
+
+import PlutusCore.Default
+import PlutusCore.Name
+import PlutusCore.TypeCheck (defKindCheckConfig)
+import PlutusCore.TypeCheck.Internal (inferKindM, runTypeCheckM, withTyVar)
+import PlutusIR
+import PlutusIR.Error
+
+instance Testable (Either String ()) where
+    property = property . \case
+        Left err -> failed { reason = err }
+        Right () -> succeeded
+
+deriving newtype instance Pretty i => Pretty (NonNegative i)
+instance PrettyBy config i => DefaultPrettyBy config (NonNegative i)
+deriving via PrettyCommon (NonNegative i)
+    instance PrettyDefaultBy config (NonNegative i) => PrettyBy config (NonNegative i)
+
+-- | Extract all @a_i@ from @a_0 -> a_1 -> ... -> r@.
+argsKind :: Kind ann -> [Kind ann]
```

Comment:
could go next to `Kind`, it's generically useful

Reply:
> could go next to `Kind`, it's generically useful

I'll do that and 

> I have something like this for Type also in the PIR compiler which should probably go in the main module

I'll let you sort out PIR compiler stuff yourself separately.